lib/config.js

```diff
@@ -50,23 +50,29 @@ const paramsSchema = Joi.object({
   auth0Logout: Joi.boolean().optional().default(false),
   authorizationParams: Joi.object({
     response_type: Joi.string().optional().valid('id_token', 'code id_token', 'code').default('id_token'),
-    scope: Joi.string().optional().default('openid profile email'),
-    response_mode: Joi.alternatives([
-      Joi.string().optional(),
-      Joi.allow(null).optional()
-    ]).default((parent) => parent.response_type.includes('token') ? 'form_post' : undefined),
+    scope: Joi.string().optional().pattern(/\bopenid\b/, 'contains openid').default('openid profile email'),
+    response_mode: Joi.string().optional().when('response_type', {
+      is: 'code',
+      then: Joi.valid('query', 'form_post').default('query'),
+      otherwise: Joi.valid('form_post').default('form_post')
+    }),
   }).optional().unknown(true).default(),
   baseURL: Joi.string().uri().required(),
   clientID: Joi.string().required(),
   clientSecret: Joi.string().when(
-    Joi.ref('authorizationParams.response_type', { adjust: (value) => value && value.split(' ').includes('code') }),
+    Joi.ref('authorizationParams.response_type', { adjust: (value) => value && value.includes('code') }),
     {
       is: true,
       then: Joi.string().required().messages({
-        'any.required': '"clientSecret" is required for response_type code'
-      }),
-      otherwise: Joi.when(
-        Joi.ref('idTokenSigningAlg', { adjust: (value) => value && value.substring(0, 2) === 'HS' }),
+        'any.required': '"clientSecret" is required for a response_type that includes code'
+      })
+    }
+  ).when(
+    Joi.ref('authorizationParams.response_type', { adjust: (value) => value && value.includes('id_token') }),
```

Comment:
`idTokenSigningAlg` being HS* is the requirement for needing a clientSecret, the response_type does not need to include `id_token` at all. ID Token will still be returned by the token endpoint with an HS signature even when response type is `code`. I believe the code before was correct and needed no changing (aside from the improved wording of the error message)

Reply:
Ah my bad, I thought that was a bug - will revert

Reply:
193b93c

After our conversation I changed the behaviour to: `clientSecret` is required if `idTokenSigningAlg` is `HS*` regardless of `response_type`